Modules/_datetimemodule.c

```diff
@@ -3087,12 +3100,19 @@ add_date_timedelta(PyDateTime_Date *date, PyDateTime_Delta *delta, int negate)
 static PyObject *
 date_add(PyObject *left, PyObject *right)
 {
-    if (PyDateTime_Check(left) || PyDateTime_Check(right))
+    _datetimemodule_state *state;
```

Comment:
This PR need a new PEP to approve this conversion.
Refs: https://www.python.org/dev/peps/pep-0573/